### DISCUSSION TITLE
cx_Freeze 'suddenly' not finding api-ms-win* libs

### DISCUSSION THREAD

**Discussion by FaceCrap** [marcelotduarte/cx_Freeze#2548] [Q&A]
Hi all,

Let me begin by saying I have a very limited knowledge of python, and even less of how cx_Freeze works. 
With that being clear, I'm running into the following problem. Now, this used to work fine but for some reason cx_Freeze is nolonger able to find the `api-ms-win*` libraries. As in, it did before, but I haven't had the need to run a build in a couple of months, I did switch to a new windows user-profile, but my memory has holes to make a sieve jealous, so no idea what might be different this time.
Searching the web I learned it looks for them using the path env variable and that I should have `C:\Windows\System32` in there. Well, obviously it is, but if I do a search with `Everything` I see these aren't in this folder but hardlinked in `C:\Windows\System32\downlevel`.
Thinking that might be the issue, I added this folder to my path env variable too, but sadly this didn't make any difference.
The original repo also provides a setup.py file, which I use to run the build process locally, but the repo owner uses a workflow to let github handle the build processs (multi-platform) and of course there it doesn't have the same issue.

This means the cause is very likely my system setup, but what? I tried excluding user-error by re-installing python (pyenv-win), recreating the project's .venv folder, reinstalling msvcrt distros (using the All-In-One package from TechPowerUp) but still no luck.
I even tried downgrading cx_Freeze to earlier versions, hoping that a change in cx_Freeze might be the reason, no luck there either.
At this point I am out of ideas.

**Comment by marcelotduarte** [comment]
Please check the [FAQ](https://cx-freeze.readthedocs.io/en/stable/faq.html#microsoft-visual-c-redistributable-package).

**Reply by FaceCrap** [reply]
> Please check the [FAQ](https://cx-freeze.readthedocs.io/en/stable/faq.html#microsoft-visual-c-redistributable-package).

I'm not totally ignorant, of course that option is fed to build_exe, otherwise it wouldn't even look for them.
But even with that option, it still does not find them where it used to be no issue before.
Even though the required msvcrt runtime is installed.

**Reply by marcelotduarte** [reply]
But the FAQ doesn't just indicate using one option, it also talks about installing the packages (either by downloading or using winget, whichever you prefer).
Maybe one possibility is the version of lief or other dependencies you're using...
I've also never tested cx_Freeze using this pyenv...

**Reply by FaceCrap** [reply]
All required packages and dependencies are installed, version of lief is whatever cx_freeze requests during the install of cx_freeze, I never install a specific dependency version if `pip install <module>` causes additional modules to be installed, because I don't want to run into situations where this might cause conflicts due to incompatibilities or differences in requirements between the respective modules. Also, there's been no changes in the build setup since it worked fine before.

>I've also never tested cx_Freeze using this pyenv...

PyEnv is just a tool to have multiple python versions installed side-by side and automate the creation of .venv environments, but I currently only have the one python version installed (3.11.2)
FYI. the resulting build runs fine on my system, which is a clear indication the required msvcrt libraries are installed and locatable. 

Don't get me wrong, I'm not saying it's a cx_Freeze/lief issue, since it used to work before, I'm just all out of ideas trying to 'fix' it, having explored all the standard options. 
I even went back as far as 6.15.13 to exclude cx_Freeze as cause, didn't go further back because 6.15.12 threw an exception `AttributeError: module 'lief._lief.logging' has no attribute 'LOGGING_LEVEL'`
So if it was a cx_Freeze issue, I'd have found a version where these dll's would be found.
Which is why I placed this in the `Discussions` area, hoping someone else had the same experience and found a solution.

**Comment by marcelotduarte** [comment]
[Release 8.0](https://github.com/marcelotduarte/cx_Freeze/releases/tag/8.0.0) is out!
[Documentation](https://cx-freeze.readthedocs.io)
